stdlib/experimental/object_keys.go

```diff
@@ -0,0 +1,38 @@
+package experimental
+
+import (
+	flux "github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+func init() {
+	flux.RegisterPackageValue("experimental", "objectKeys", values.NewFunction(
+		"objectKeys",
+		semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+			Parameters: map[string]semantic.PolyType{
+				"o": semantic.Tvar(1),
+			},
+			Required: []string{"o"},
+			Return:   semantic.NewArrayPolyType(semantic.String),
+		}),
+		func(args values.Object) (values.Value, error) {
+			o, ok := args.Get("o")
+			if !ok {
+				return nil, errors.New(codes.Invalid, "missing parameter \"o\"")
+			}
+			if o.Type().Nature() != semantic.Object {
+				return nil, errors.New(codes.Invalid, "parameter \"o\" is not an object")
+			}
+			obj := o.Object()
+			keys := make([]values.Value, 0, obj.Len())
+			obj.Range(func(name string, _ values.Value) {
+				keys = append(keys, values.NewString(name))
```

Comment:
I wonder if this will return the keys in any particular order, or if determinism might be a concern here?

Reply:
The implementation is deterministic and uses the order they are defined. I thought that was part of the SPEC, but a second look doesn't mention it.